src/data_collector/polygon_fundamentals_v2/validator_v2.py

```
@@ -1,88 +1,438 @@
+            # Validate string fields for injection attacks
+            if self.unit:
+                SecureString.validate_secure_string(self.unit)
+            if self.label:
+                SecureString.validate_secure_string(self.label)
+            if self.source:
+                SecureString.validate_secure_string(self.source)
+
+            # Validate order as reasonable integer
+            if self.order is not None:
+                if not isinstance(self.order, int) or self.order < 0 or self.order > 1000:
+                    raise SecurityValidationError("Invalid order value", "order", self.order)
```

Ruff (0.14.3)

Abstract `raise` to an inner function

(TRY301)